### PR TITLE
Retry when docker push fails

### PR DESCRIPTION
**Pull request by withsmilo**
* related issue : no.7 in https://github.com/ucbrise/clipper/issues/656
* I imported a Retry decorator from https://www.saltycrane.com/blog/2009/11/trying-out-retry-decorator-python.

### CODE REVIEW

**Comment by simon-mo** [comment]
This looks great. We have several cases of sleep and retry in clipper_admin, do you want to refactor them as well in this PR?

**Comment by withsmilo** [comment]
@simon-mo 
Yep, I will try to refactor [these codes](https://github.com/ucbrise/clipper/blob/develop/clipper_admin/clipper_admin/clipper_admin.py#L127-L142).

File: clipper_admin/clipper_admin/clipper_admin.py [hunk 1]
```diff
@@ -124,28 +126,31 @@ def start_clipper(self,
...
-                    self.logger.info("Clipper still initializing: \n {}".format(e))
-                    time.sleep(1)
-            self.logger.info("Clipper is running")
-            self.connected = True
         except ClipperException as e:
             self.logger.warning("Error starting Clipper: {}".format(e.msg))
             raise e
 
+        # Wait for maximum 5 min.
+        @retry(RequestException, tries=300, delay=1, backoff=1, logger=self.logger)
```
Comment (withsmilo) [inline clipper_admin/clipper_admin/clipper_admin.py:134]: @simon-mo 
Is this value suitable?

Comment (simon-mo) [reply clipper_admin/clipper_admin/clipper_admin.py:134]: Yes. 5min is reasonable. We might want to increase the interval later if it prints too much intonations to the console.  However, current interval looks good enough.

**Review by simon-mo — APPROVED** [review]

**Pull request merged**